### PR TITLE
Private_events

### PR DESCRIPTION
**Pull request by MarkoNS1990**
🟢 In this project we created a simple website similar to Eventbrite where users login before creating events. Users can also attend other users events. 🟢

### CODE REVIEW

**Review by mricanho — CHANGES_REQUESTED** [review]
# 🔴  Changes Required 🔴  

Hello @MarkoNS1990 @mgunduz1, you did a great job with your private events, the website it's fully functional and that's something that we need to celebrate🔥, just a few changes are required:

### General
- [x]  **There's a bug, when an event is completed(past event), and a random user open the event, the user can join the event even if this is a past event, kindly fix this** 
- [x] **The "Home" navigation link must be present all the time in order to return to the main page at any moment**

### Controllers
- [x] **In the create action, you need to use #[build ](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/build_method_requirement.png)method on association, i.e. `current_user.events.build(event_params)`**
- [x] **You are taking the expired and future events from the scope to display them in the views, but you actually need to create in the [Index ](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png)action a lists past events(or expired_events), i.e. `@past_events = Event.previous_events` the same with future events**

### Tests

- [x] **You need to create tests for your [models](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png), "models have unit tests for validations and associations"**
- [x] **Double check the configuration of your RSpec, right now none of the tests are running**

It's a lot, but you are so close to finish this project, and it was a really good first try, you got this.

you can do it


You can contact me on slack if you have any questions just search for me as @ Miguel Ricaño.

### 💻 👩🏽‍💻  Happy Coding!👨🏽‍💻  🖥️

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    <% if user_signed_in? %>
+    <%= link_to 'Logout', destroy_user_session_path, method: :delete, class:"btn btn-lg btn-danger text-white" %>
+    <%= link_to 'New Event', new_event_path, class:"btn btn-lg btn-success text-white" %>
+    <%= link_to 'My profile', user_path(current_user), class:'btn btn-lg btn-primary text-white' %>
+    <% else %>
+    <%= link_to 'Register', new_user_registration_path, class:"btn btn-lg btn-success text-white" %>
+    <%= link_to 'Login', new_user_session_path, class:"btn btn-lg btn-primary text-white" %>
+
+    <% end %>
```
Comment (enaburekhan) [inline app/views/layouts/application.html.erb:25]: **Optional**

- [ ] Move this logic to a helper. Views are only for rendering what's made available to them from the corresponding controller. And not for generating it.

File: app/views/users/profile.html.erb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        <div class="card-body">
+          <%= event.event.description %>
+        </div>
+        <div class="card-footer d-flex justify-content-between">
+          <p class="fs-6 text-dark">Created By: <%= event.event.user.email %></p>
+          <p class="fs-6 text-dark">Event Date: <%= event.event.date %></p>
+        </div>
+      </div>
+    </div>
+  <% end %>
```
Comment (enaburekhan) [inline app/views/users/profile.html.erb:73]: Also, move this logic to a helper.

**Review by enaburekhan — APPROVED** [review]
# Status: Approve:heavy_check_mark:
**Hi team!
Congratulations:clap::clap:. Thank you for implementing all the changes requested by the last reviewer.
Before you go ahead and merge the PR. I will encourage you to implement the `Optional changes` highlighted in the PR below. This will make the project codebase looks more professional.**

Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.

**Pull request merged**